pkg/asset-manager-utils/test/AaveATokenAssetManager.test.ts

```diff
@@ -377,83 +378,140 @@ describe('Aave Asset manager', function () {
+        });
+
+        it('transfers the expected number of tokens to the Vault', async () => {
+          const { poolCash, poolManaged } = await assetManager.getPoolBalances(poolId);
+          const expectedRebalanceAmount = calcRebalanceAmount(poolCash, poolManaged, poolConfig);
+
+          await expectBalanceChange(() => assetManager['rebalance(bytes32,bool)'](poolId, forceRebalance), tokens, [
+            { account: lendingPool.address, changes: { DAI: expectedRebalanceAmount } },
+            { account: vault.address, changes: { DAI: expectedRebalanceAmount.mul(-1) } },
+          ]);
+        });
+
+        it('returns the pool to its target allocation', async () => {
+          await assetManager['rebalance(bytes32,bool)'](poolId, forceRebalance);
+          const differenceFromTarget = await assetManager.maxInvestableBalance(poolId);
+          expect(differenceFromTarget.abs()).to.be.lte(1);
+        });
+
+        it("updates the pool's cash and managed balances correctly");
+      } else {
+        it('skips the rebalance', async () => {
+          const tx = await assetManager['rebalance(bytes32,bool)'](poolId, forceRebalance);
+          const receipt = await tx.wait();
+          expectEvent.notEmitted(receipt, 'Rebalance');
+        });
+      }
+    }
+
+    const poolConfig = {
+      targetPercentage: fp(0.5),
+      upperCriticalPercentage: fp(0.75),
+      lowerCriticalPercentage: fp(0.25),
+    };
+
+    sharedBeforeEach(async () => {
+      const poolController = lp; // TODO
+      await assetManager.connect(poolController).setPoolConfig(poolId, poolConfig);
+    });
+
     context('when pool is above target investment level', () => {
-      let poolController: SignerWithAddress; // TODO
-      const poolConfig = {
-        targetPercentage: fp(0.5),
-        upperCriticalPercentage: fp(1),
-        lowerCriticalPercentage: fp(0.1),
-      };
+      context('when pool is in non-critical range', () => {
+        sharedBeforeEach(async () => {
+          const poolController = lp; // TODO
+          await assetManager.connect(poolController).setPoolConfig(poolId, poolConfig);
+          const amountToDeposit = await assetManager.maxInvestableBalance(poolId);
+          await assetManager.connect(poolController).capitalIn(poolId, amountToDeposit);
+
+          // should be perfectly balanced
+          const maxInvestableBalance = await assetManager.maxInvestableBalance(poolId);
+          expect(maxInvestableBalance).to.equal(bn(0));
+
+          // Simulate a return on asset manager's investment
+          const amountReturned = amountToDeposit.div(10);
+          await lendingPool
+            .connect(lp)
+            .simulateATokenIncrease(tokens.DAI.address, amountReturned, assetManager.address);
+        });
+
+        context('when forced', () => {
+          itRebalancesCorrectly(true, true);
+        });
+
+        context('when not forced', () => {
+          itRebalancesCorrectly(false, false);
+        });
```

Comment:
```suggestion
        context('when forced', () => {
          const forced = true;
          itRebalancesCorrectly(true, forced);
        });

        context('when not forced', () => {
          const forced = false;
          itRebalancesCorrectly(false, forced);
        });
```

Reply:
(this test is quite simple, but this pattern is very handy when the setting up of the test is a bit more involved and e.g. the `= true` would happen inside a `beforeEach`